### PR TITLE
Deprecate `spike_dilutor` and prohibit its use with multiple threads

### LINKED ISSUE #2052: Issue with Spike Dilutor and Multi-Threading
**Describe the bug**
spike_dilutor behavior changes when using multiple threads. 

**To Reproduce**
Here's a small example that exhibits the behavior:
```
import numpy
import nest

# Set Threads https://nest-test.readthedocs.io/en/exhale_test/guides/parallel_computing.html#using-multiple-threads
T=4
nest.SetKernelStatus({"local_num_threads":T})


# Set PRNG seeds https://nest-test.readthedocs.io/en/exhale_test/guides/random_numbers.html#random-numbers-in-parallel-simulations
msd = 123456
N_vp = nest.GetKernelStatus(['total_num_virtual_procs'])[0]
pyrngs = [numpy.random.RandomState(s) for s in range(msd, msd+N_vp)]
nest.SetKernelStatus({'grng_seed' : msd+N_vp})
nest.SetKernelStatus({'rng_seeds' : range(msd+N_vp+1, msd+2*N_vp+1)})

# Define source neurons
neuron_list = []
neuron1 = nest.Create("iaf_psc_delta")
nest.SetStatus(neuron1, {"I_e": 376.0})
neuron_list.append(neuron1)

#Define dilutor with 50% copy
neuron2 = nest.Create("spike_dilutor",params={'p_copy':0.5})

#Create and connect recorders; connect source neurons and dilutor
recorder = nest.Create("spike_recorder")
source_recorder = nest.Create("spike_recorder")

for n in neuron_list:
  nest.Connect(n, neuron2)
  nest.Connect(n, source_recorder)
nest.Connect(neuron2, recorder)

#Run
nest.Simulate(10000)

source_spikes = nest.GetStatus(source_recorder)[0]
spikes = nest.GetStatus(recorder)
print("Original spikes: {}".format(source_spikes))
print("Diluted spikes: {}".format(spikes))
```

- When T = 1, we see 163 source spikes and 77 spikes from the dilutor (expected)
- When T = 2, we see 163 source spikes but 87 spikes from the dilutor
- When T= 4, we see 163 source spikes but 0 spikes from the dilutor
- When T= 8, we see 163 source spikes but 0 spikes from the dilutor

**Expected behavior**
Dilutor should remain consistent, i.e. relay p_copy of the spikes (in this case, 50%).

**Desktop/Environment (please complete the following information):**
 - OS: Ubuntu 20.4.1 LTS
 - Shell: bash
 - Python-Version: 3.8.5
 - NEST-Version: NEST version master@7611bede8, built on Nov 12 2020 17:49:12 for x86_64-pc-linux
 - Installation: Docker

**Additional context**
From the full-scale network, the problem seems to get worse as the number of threads increases.  Issue persists on both Intel Core i9 and AMD Epyc processors.

### PR DESCRIPTION
**Pull request by heplesser**
`spike_dilutor` is a variant of a parrot neuron which probabilistically forwards spikes. Each target neuron receives a different thinned spike train. Therefore, `spike_dilutor` needs to send `DSSpikeEvent`s and is implemented as a `StimulationDevice`. This conflicts, though, with the fact that the node also needs to receive spikes, leading to incorrect behavior when used with multiple threads (#2052).

At the same time, `bernoulli_synapse` implements exactly the same functionality in a proper, i.e., synapse, not pseudo-neuron, based way. This PR therefore

1. deprecates `spike_dilutor`
2. prohibits use of `spike_dilutor` with multiple threads
3. points to `bernoulli_synapse` as alternative.

This fixes #2052.

### CODE REVIEW

File: testsuite/regressiontests/issue-2052.sli [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ */
+
+
+ /** @BeginDocumentation
+Name: testsuite::issue-2052
+
+Synopsis: (issue-2052) run -> NEST exits if test fails
+
+Description:
+Ensure that spike_dilutor cannot be created with a single but not multiple threads.
```
Comment (hakonsbm) [inline testsuite/regressiontests/issue-2052.sli:30]:
```suggestion
Ensure that spike_dilutor can be created with a single thread, but not with multiple threads.
```

**Review by hakonsbm — CHANGES_REQUESTED** [review]
Looks good to me, except for a minor clarification in the test documentation.

**Review by hakonsbm — APPROVED** [review]
Thanks!

**Review by jougs — APPROVED** [review]
Thanks!

**Comment by jougs** [comment]
@heplesser: can you please fix the conflicts? Thanks!

**Pull request merged**